internal/pkg/runtime/engine/singularity/prepare_linux.go

```diff
@@ -1166,9 +1166,16 @@ func (e *EngineOperations) loadImages(starterConfig *starter.Config) error {
 				return fmt.Errorf("while validating ECL configuration: %s", err)
 			}
 
-			kr, err := sypgp.PublicKeyRing()
-			if err != nil {
-				return fmt.Errorf("while obtaining keyring for ECL: %s", err)
+			// Only try to load the keyring here if the ECL is active.
+			// Otherwise pass through an empty keyring rather than avoiding calling
+			// the ECL functions as this keeps the logic for applying / ignoring ECL in a
+			// single location.
+			kr := sypgp.NullKeyRing()
+			if ecl.Activated {
```

Comment:
Arguably, if this is `false`, we don't need to call `ecl.ShouldRunFp` at all, and thus wouldn't need a keyring? What about something like: 

```
			if ecl.Activated {
				kr, err := sypgp.PublicKeyRing()
				if err != nil {
					return fmt.Errorf("while obtaining keyring for ECL: %s", err)
				}

				if ok, err := ecl.ShouldRunFp(img.File, kr); err != nil {
					return fmt.Errorf("while checking container image with ECL: %s", err)
				} else if !ok {
					return errors.New("image prohibited by ECL")
				}
			}
```

Reply:
Now modified per discussion elsewhere,